docs_src/manuals/user_manual/execution_model.hrst

```diff
@@ -31,15 +31,105 @@ execution policy. A computation combines several multistages and executes one mu
   previous stages of the multistage have not already been applied to the indices in the same column with smaller ``k``.
 
 
+----------------------------------
+Extended Compute Domain of a Stage
+----------------------------------
+
+For a clear specification of access restrictions, we need to introduce the concept of extended compute domains.
+
+If a stage is accessing neighbor grid points of field `a`, the computation will consume halo points at the boundary of the compute domain.
+In case field `a` was written in a previous stage of the same multi-stage, the compute domain of the first stage needs to be extended, such
+that the *extended compute domain* of the stage covers the extent of the read access in the later stage.
+
 -------------------
 Access restrictions
 -------------------
 
-The execution model imposes restrictions on how accessors can be evaluated. The following restrictions apply:
+The execution model imposes restrictions on how accessors can be evaluated. These restrictions are not checked by GridTools,
+violating these rules results in undefined behavior.
+
+Definitions:
+
+ - *Writing/reading to/from an accessor* is short for *writing/reading to/from a field through an accessor*.
+   Note that the same field could be bound to different accessors, though this is not recommended.
+
+- *Previous/later statement* is short for *previous/later statement in the same stage or any previous/later stage*.
+
+
+The following rules apply to accesses in stages *within the same multi-stage*:
+
+ 1. A stage may write to an accessor, if there is no read from the same accessor in a previous statement.
```

Comment:
So we want to forbid write-after-read. But this makes perfectly sense in some cases, think about a simple scaling stencil (focus on x)
```
Stage 1:
    tmp = x
    x = tmp * 2
```
This is only a problem if all this happens in an extended compute domain, and for non-temporaries.

Reply:
Why do temporaries change the game here?

Reply:
```
Stage 1:
    tmp = x
    x = tmp *2
Stage 2:
    y = x(i+1)
```

Reply:
if x is temporary, the extended compute domain is redundant. If we have two blocks doing the redundant computation, it happens in diferent memory locations. If x is non'temporary, the memory location of the redundant computation is the same.

Reply:
and only if the memory location is the same, you can get in trouble that different blocks mess up with read-write dependencies.